### ISSUE TITLE
Add support for the `name:` directive 

### ISSUE THREAD

**Issue by MartinHammarstedt** [JetBrains-Research/snakecharm#351]
Please add support for the `name:` directive added in [Snakemake 5.31.0](https://github.com/snakemake/snakemake/blob/main/CHANGELOG.rst#5310---2020-12-21).

```
rule:
    name: rule_name_variable
```

**Comment by iromeo** [comment]
Usage example:

```
rule all:
    input:
        txt = 'out2.txt'

some_var: str = 'generate_out'

rule:
    name: some_var
    output: 'out1.txt'
    shell: '''echo {rule} > {output}'''


rule name1:
    name: "name2"
    input: rules.generate_out.output
    output: 'out2.txt'
    shell: '''echo {rule} > {output}'''
```

**Comment by iromeo** [comment]
@MartinHammarstedt thx for the feature request, I'll add some initial support, at list syntax one. But what is a real benefit and use case of this `name:` directive?

**Comment by MartinHammarstedt** [comment]
The main benefit of the `name:` directive is being able to generate named rules programmatically. Like this:

```python

def create_rule(rule_name, ...):
    rule:
        name: rule_name
        ...
```

**Comment by iromeo** [comment]
@MartinHammarstedt thx, and why do you need to define them programmatically? Is it just to avoid code duplication in similar rules / add more dynamic features into them? Or it is some code-generation tool for pipelines?

**Comment by MartinHammarstedt** [comment]
In my case it's... complicated. :) We've built a pipeline where you create all the different steps of the pipeline using pure Python, which are then automatically translated into Snakemake rules during runtime, and that's where `name:` is needed.

**Comment by iromeo** [comment]
I don't use code generation patterns in my pipeline, so I'm the plugin is not tested for cases when rules declarations are in methods, loops, etc. Feel free to report about things which doesn't work in such use case.

As for this issue, I've just added initial parsing/syntax highlighting for this as part of #334. I.e. no errors in code + 'name' will be in auto-completion. Not sure about whether we need smart support, e.g. somehow consider this dynamic rule names in `rules.<carret>` autocompletion. Or understand that rule `name1` is actually `name2` on runtime in the example below:

![image](https://user-images.githubusercontent.com/72933/115956377-01e2ea80-a505-11eb-91eb-3b7df55f768e.png)

So support of `name:` will be included on syntax level in nearest major update. As for smarter features, I'm going to it to a backlog.

**Comment by iromeo** [comment]
done, will be in next release (0.10.x)

**Comment by iromeo** [comment]
@MartinHammarstedt The new plugin with the fixed has been released, it is compatible with PyCharm 2021.1.x, see https://twitter.com/iRomeo_work/status/1386031601927213057?s=20

**Comment by MartinHammarstedt** [comment]
That was quick! Thank you!
